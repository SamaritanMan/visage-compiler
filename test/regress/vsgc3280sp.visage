/**
 * VSGC-3280 : override var x on replace... fails in a Stage initializer
 *
 * @test
 * @run
 */

class Nod {
   var spx: Object = null  on replace {
       println("spx on replace called for Nod {spx}");
   }
   public var px: Object = null  on replace {
       println("px on replace called for Nod {px}");
   }
   var bx: Object = null  on replace {
       println("bx on replace called for Nod {bx}");
   }
   def bbx = bind bx;

   var sps: String = "" on replace {
       println("sps on replace called for Nod {sps}");
   }
   public var ps: String = "" on replace {
       println("ps on replace called for Nod {ps}");
   }
   var bs: String = "" on replace {
       println("bs on replace called for Nod {bs}");
   }
   def bbs = bind bs;

   var spn: Number = 0.0 on replace {
       println("spn on replace called for Nod {spn}");
   }
   public var pn: Number = 0.0  on replace {
       println("pn on replace called for Nod {pn}");
   }
   var bn: Number = 0.0  on replace {
       println("bn on replace called for Nod {bn}");
   }
   def bbn = bind bn;

   var spxv: Object = "rah"  on replace {
       println("spxv on replace called for Nod {spxv}");
   }
   public var pxv: Object = "rah"  on replace {
       println("pxv on replace called for Nod {pxv}");
   }
   var bxv: Object = "rah"  on replace {
       println("bxv on replace called for Nod {bxv}");
   }
   def bbxv = bind bxv;

   var spsv: String = "hello" on replace {
       println("spsv on replace called for Nod {spsv}");
   }
   public var psv: String = "hello" on replace {
       println("psv on replace called for Nod {psv}");
   }
   var bsv: String = "hello" on replace {
       println("bsv on replace called for Nod {bsv}");
   }
   def bbsv = bind bsv;

   var spnv: Number = 12.3 on replace {
       println("spnv on replace called for Nod {spnv}");
   }
   public var pnv: Number = 12.3  on replace {
       println("pnv on replace called for Nod {pnv}");
   }
   var bnv: Number = 12.3  on replace {
       println("bnv on replace called for Nod {bnv}");
   }
   def bbnv = bind bnv;

}

class Grup extends Nod {

    override var spx on replace {
       println("spx on replace called for Grup {spx}");
    }
    override var px on replace {
       println("px on replace called for Grup {px}");
    }
    override var bx on replace {
       println("bx on replace called for Grup {bx}");
    }

    override var sps on replace {
       println("sps on replace called for Grup {sps}");
    }
    override var ps on replace {
       println("ps on replace called for Grup {ps}");
    }
    override var bs on replace {
       println("bs on replace called for Grup {bs}");
    }

    override var spn on replace {
       println("spn on replace called for Grup {spn}");
    }
    override var pn on replace {
       println("pn on replace called for Grup {pn}");
    }
    override var bn on replace {
       println("bn on replace called for Grup {bn}");
    }

    override var spxv on replace {
       println("spxv on replace called for Grup {spxv}");
    }
    override var pxv on replace {
       println("pxv on replace called for Grup {pxv}");
    }
    override var bxv on replace {
       println("bxv on replace called for Grup {bxv}");
    }

    override var spsv on replace {
       println("spsv on replace called for Grup {spsv}");
    }
    override var psv on replace {
       println("psv on replace called for Grup {psv}");
    }
    override var bsv on replace {
       println("bsv on replace called for Grup {bsv}");
    }

    override var spnv on replace {
       println("spnv on replace called for Grup {spnv}");
    }
    override var pnv on replace {
       println("pnv on replace called for Grup {pnv}");
    }
    override var bnv on replace {
       println("bnv on replace called for Grup {bnv}");
    }
}

def gg = Grup {};

gg.spx = "sp-yo";
gg.px = "p-yo";
gg.bx = "b-yo";

gg.sps = "sp-hi";
gg.ps = "p-hi";
gg.bs = "b-hi";

gg.spn = 1;
gg.pn = 2;
gg.bn = 3;

gg.spxv = "sp-yo";
gg.pxv = "p-yo";
gg.bxv = "b-yo";

gg.spsv = "sp-hi";
gg.psv = "p-hi";
gg.bsv = "b-hi";

gg.spnv = 1;
gg.pnv = 2;
gg.bnv = 3;


