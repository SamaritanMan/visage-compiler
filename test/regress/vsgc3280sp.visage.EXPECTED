spx on replace called for Nod null
spx on replace called for Grup null
px on replace called for Nod null
px on replace called for Grup null
bx on replace called for Nod null
bx on replace called for Grup null
sps on replace called for Nod 
sps on replace called for Grup 
ps on replace called for Nod 
ps on replace called for Grup 
bs on replace called for Nod 
bs on replace called for Grup 
spn on replace called for Nod 0.0
spn on replace called for Grup 0.0
pn on replace called for Nod 0.0
pn on replace called for Grup 0.0
bn on replace called for Nod 0.0
bn on replace called for Grup 0.0
spxv on replace called for Nod rah
spxv on replace called for Grup rah
pxv on replace called for Nod rah
pxv on replace called for Grup rah
bxv on replace called for Nod rah
bxv on replace called for Grup rah
spsv on replace called for Nod hello
spsv on replace called for Grup hello
psv on replace called for Nod hello
psv on replace called for Grup hello
bsv on replace called for Nod hello
bsv on replace called for Grup hello
spnv on replace called for Nod 12.3
spnv on replace called for Grup 12.3
pnv on replace called for Nod 12.3
pnv on replace called for Grup 12.3
bnv on replace called for Nod 12.3
bnv on replace called for Grup 12.3
spx on replace called for Nod sp-yo
spx on replace called for Grup sp-yo
px on replace called for Nod p-yo
px on replace called for Grup p-yo
bx on replace called for Nod b-yo
bx on replace called for Grup b-yo
sps on replace called for Nod sp-hi
sps on replace called for Grup sp-hi
ps on replace called for Nod p-hi
ps on replace called for Grup p-hi
bs on replace called for Nod b-hi
bs on replace called for Grup b-hi
spn on replace called for Nod 1.0
spn on replace called for Grup 1.0
pn on replace called for Nod 2.0
pn on replace called for Grup 2.0
bn on replace called for Nod 3.0
bn on replace called for Grup 3.0
spxv on replace called for Nod sp-yo
spxv on replace called for Grup sp-yo
pxv on replace called for Nod p-yo
pxv on replace called for Grup p-yo
bxv on replace called for Nod b-yo
bxv on replace called for Grup b-yo
spsv on replace called for Nod sp-hi
spsv on replace called for Grup sp-hi
psv on replace called for Nod p-hi
psv on replace called for Grup p-hi
bsv on replace called for Nod b-hi
bsv on replace called for Grup b-hi
spnv on replace called for Nod 1.0
spnv on replace called for Grup 1.0
pnv on replace called for Nod 2.0
pnv on replace called for Grup 2.0
bnv on replace called for Nod 3.0
bnv on replace called for Grup 3.0
